File: c/hackerank/dynamic-array.c
#include <stdio.h>
#include <malloc.h>


typedef struct dynamic_arry_t {
    int *array;
    unsigned long long int len;
    unsigned int alloc_len;
} dynamic_array_t;


void append(dynamic_array_t *array){
    if(array->alloc_len%10 == 0){
        array->alloc_len += 10 
        realloc(array->array, array->alloc_len);
    }
}

int main(){
    int n, lastAns = 0, queries;
    scanf('%d %d', &n, &queries);
    dynamic_array_t *seqlist[n];

    for(int query=0; query<queries; ++query){
        
    }
}

File: python/hackerank/dynamic-array.py
lastAns = 0

n, queries = map(int, input().split())

seqlist = [[] for _ in range(n)]

for _ in range(queries):
    query = list(map(int, input().split()))
    ans = (int(query[1]) ^ lastAns) % n
    # print('Calculaion now', ans)

    if query[0] == 1:
        seqlist[ans].append(int(query[2]))
        # print('Current list', seqlist)
    
    elif query[0] == 2:
        # print(seqlist[ans])
        lastAns = seqlist[ans][int(query[2])%len(seqlist[ans])]
        print(lastAns)